### ISSUE TITLE
DelaTable function returns columns as per file write schema, not as per merged schema

### ISSUE THREAD

**Issue by AkshayTilekar** [jeppe742/DeltaLakeReader#26]
```
from deltalake import DeltaTable
from s3fs import S3FileSystem

fs = S3FileSystem()  # authenticate using environment variables, in this example
df = DeltaTable(f"s3://{DATA_BUCKET}/{RES_PATH}", file_system=fs)
res = df.to_table(filter= (ds.field("s_uid") == str(s_uid)) & (ds.field("r_uid") == str(r_uid))).to_pandas()

```
I am reading a delta table from AWS S3 using above function.
and I did three writes to the delta table;

1. Write table with 4 columns.
2. Write table with 1 extra column. Now table has 5 columns.
3. Write table again with only 4 columns which we did not have extra column 1 above.

Now, When I read a delta table it only shows 4 columns, instead it should show all the 5 columns.

Any help ?

**Comment by jeppe742** [comment]
Hey @AkshayTilekar,

I actually wasn't aware you could do this.
If you read the table using spark, do you get all 5 columns? And are the missing column from the third write just filled with NULLs?

**Comment by AkshayTilekar** [comment]
Yes, Exactly. In Spark it is working fine. 
I've seen the DeltaTable code and what I found is, DeltaTable is referring to the latest file to generate the schema and in my case that latest file is having fewer columns than the actual. I guess we need to have something which iterates over the files and creates a superset schema considering all columns.

Will love to debug more and contribute.

**Comment by jeppe742** [comment]
Hmm, think I saw somewhere that the schema could be specified in the delta log file.
I guess nothing is stopping you from fixing it, but otherwise I might try to take a look at it later during the weekend :)

**Comment by AkshayTilekar** [comment]
Cool, Thanks.

**Comment by jeppe742** [comment]
Hey @AkshayTilekar.
Took a bit longer than expected, but think I found a way of getting the expected schema explicitly from the delta logs.
This should fix your issue.

Created the fix in #27, but since there currently seems to be an outage in github actions I will wait merging it until the build have managed to run.

**Comment by jeppe742** [comment]
Merged and published new package with version 0.2.8

**Comment by AkshayTilekar** [comment]
Thank you 😀 
Let me check with it.
